src/librustc/ty/query/plumbing.rs

```diff
@@ -139,39 +149,54 @@ impl<'a, 'tcx, Q: QueryDescription<'tcx>> JobOwner<'a, 'tcx, Q> {
                                 query_blocked_prof_timer = Some(tcx.prof.query_blocked());
                             }
 
-                            job.clone()
+                            // Create the id of the job we're waiting for
+                            let id = QueryJobId {
```

Comment:
Would you mind creating a helper method for this? This function is already rather big.

Reply:
Done.